multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/jobs/CleanUpJob.java

```diff
@@ -31,8 +30,11 @@
     List<Cleaner> cleaners;
     private final SafeExecutor safeExecutor = new SafeExecutor(CleanUpJob::log);
 
-    @Override
-    public void execute(JobExecutionContext context) {
+    @Scheduled(cron = "#{@applicationConfiguration.getCronExpressionForOldData()}")
+    public void execute() {
+        if (configuration.getApplicationInstanceIndex() != 0) {
+            return;
```

Comment:
We want the cleaners to run on only one instance, because the clean up job. With quartz, we used the annotation `@DisallowConcurrentExecution`, which enables locking mechanism in the DB.